ouroboros-consensus-cardano/tools/db-analyser/Block/Alonzo.hs

```diff
@@ -0,0 +1,91 @@
+{-# LANGUAGE DeriveAnyClass             #-}
+{-# LANGUAGE DerivingStrategies         #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE NamedFieldPuns             #-}
+{-# LANGUAGE OverloadedStrings          #-}
+{-# LANGUAGE StandaloneDeriving         #-}
+
+{-# OPTIONS_GHC -Wno-orphans #-}
+module Block.Alonzo () where
+
+-- * Prelude
+import           Prelude
+-- * Aeson
+import           Data.Aeson (FromJSON (..), (.:), (.:?))
+import qualified Data.Aeson as Aeson
+import           Data.Aeson.Types (FromJSONKey (..))
+-- * Alonzo
+import qualified Cardano.Ledger.Alonzo.Genesis as Alonzo
+import qualified Cardano.Ledger.Alonzo.Language as Alonzo
+import qualified Cardano.Ledger.Alonzo.Scripts as Alonzo
+import qualified Cardano.Ledger.BaseTypes as Ledger
+-- * Plutus
+import           Plutus.V1.Ledger.Api (defaultCostModelParams)
+-- * Other
+import           Control.Applicative
+import qualified Data.Map.Strict as Map
+import           Data.Text (Text)
+
+instance FromJSON Alonzo.AlonzoGenesis where
+  parseJSON = Aeson.withObject "Alonzo Genesis" $ \o -> do
+    coinsPerUTxOWord     <- o .:  "lovelacePerUTxOWord"
+                        <|> o .:  "adaPerUTxOWord"
+    cModels              <- o .:? "costModels"
+    prices               <- o .:  "executionPrices"
+    maxTxExUnits         <- o .:  "maxTxExUnits"
+    maxBlockExUnits      <- o .:  "maxBlockExUnits"
+    maxValSize           <- o .:  "maxValueSize"
+    collateralPercentage <- o .:  "collateralPercentage"
+    maxCollateralInputs  <- o .:  "maxCollateralInputs"
+    case cModels of
```

Comment:
How about splitting this into:

```
costmdls <- case cModels ...
return Alonzo.AlonzoGenesis
    { Alonzo.coinsPerUTxOWord
    , Alonzo.collateralPercentage
    , Alonzo.costmdls
    , Alonzo.maxBlockExUnits
    , Alonzo.maxCollateralInputs
    , Alonzo.maxTxExUnits
    , Alonzo.maxValSize
    , Alonzo.prices
    }
```

Reply:
I hope to get this done input-output-hk/cardano-ledger-specs#2452 (if ledger folks give me 👍 ) but till now I want to keep this code 1:1 with what cardano-node is having. How it should look I would love to address on the PR that will implement input-output-hk/cardano-ledger-specs#2452. Does it make sense?

Reply:
Makes sense. (I see that 2452 there has garnered sufficient thumbs-ups! 🙌.)